src/NumericalAlgorithms/Interpolation/CubicSpline.cpp

```diff
@@ -0,0 +1,66 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "NumericalAlgorithms/Interpolation/CubicSpline.hpp"
+
+#include <algorithm>
+#include <cstddef>
+#include <gsl/gsl_spline.h>
+#include <memory>
+#include <pup.h>
+
+#include "ErrorHandling/Assert.hpp"
+
+namespace intrp {
+CubicSpline::CubicSpline(std::vector<double> x_values,
+                         std::vector<double> y_values) noexcept
+    : x_values_(std::move(x_values)), y_values_(std::move(y_values)) {
+  ASSERT(x_values_.size() == y_values_.size(),
+         "The x-value and y-value vectors must be of the same length, but "
+         "received x-value of size: "
+             << x_values_.size()
+             << " and y-value of size: " << y_values_.size());
+  ASSERT(std::is_sorted(x_values_.begin(), x_values_.end()),
+         "The x-values must be sorted.");
+  initialize_interpolant();
+}
+
+void CubicSpline::gsl_interp_accel_deleter::operator()(
+    gsl_interp_accel* acc) const noexcept {
```

Comment:
This was commented on before, but doesn't appear to have been done: This argument can be const. (Const pointer, not pointer to const.)  (And again below.)

Reply:
@wthrowe did I do anything wrong when I tried that here https://github.com/sxs-collaboration/spectre/pull/1677#discussion_r323336715 ? I ran into trouble with `const gsl_interp_accel* acc`.

Reply:
That's a pointer to const, not a const pointer.

Reply:
thanks, fixed